lib/src/Uncertainty/Algorithm/OrthogonalBasis/OrthogonalProductPolynomialFactory.cxx

```diff
@@ -304,4 +304,16 @@ Sample OrthogonalProductPolynomialFactory::getNodesAndWeights(const Indices & de
   return nodes;
 }
 
+/* Get marginal functions */
+OrthogonalProductPolynomialFactory::PolynomialFamilyCollection OrthogonalProductPolynomialFactory::getMarginal(const Indices & indices) const
+{
+  const UnsignedInteger size = coll_.getSize();
+  indices.check(size);
```

Comment:
This does not throw an exception if necessary: it returns a boolean. Please throw an exception if necessary.

Reply:
Fixed at c3d2700be4acea9bbc0c1ea1e8f6316948ac3e4e